fv3config/config/diag_table.py

```diff
@@ -0,0 +1,212 @@
+import logging
+from typing import Sequence, Optional, Union
+import dataclasses
+import datetime
+import re
+
+import dacite
+
+from .._exceptions import ConfigError
+
+logger = logging.getLogger("fv3config")
+
+
+@dataclasses.dataclass
+class DiagTableField:
+    """Object representing a field of a diagnostics file."""
+
+    module_name: str
+    field_name: str
+    output_name: str
+    reduction_method: Optional[str] = "none"
+    time_sampling: Optional[str] = "all"
+    regional_section: Optional[str] = "none"
+    packing: Optional[int] = 2
+
+
+@dataclasses.dataclass
+class DiagTableFile:
+    """Object representing a diagnostics file."""
+
+    name: str
+    frequency: int
+    frequency_units: str
+    fields: Sequence[DiagTableField]
+    file_format: Optional[int] = 1
+    time_axis_units: Optional[str] = "hours"
+    time_axis_name: Optional[str] = "time"
```

Comment:
Is this referring to a netCDF coordinate? I'm supportive of keeping the same naming as in the diag_table documentation, but you may want to add a docstring if you know the answer to this.

Reply:
Sure I can add a docstring/comment.